htlcswitch/link_test.go

```diff
@@ -0,0 +1,615 @@
+package htlcswitch
+
+import (
+	"bytes"
+	"fmt"
+	"testing"
+	"time"
+
+	"reflect"
+
+	"io"
+
+	"github.com/davecgh/go-spew/spew"
+	"github.com/go-errors/errors"
+	"github.com/lightningnetwork/lnd/lnwallet"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"github.com/lightningnetwork/lnd/routing"
+	"github.com/roasbeef/btcd/chaincfg/chainhash"
+	"github.com/roasbeef/btcd/wire"
+	"github.com/roasbeef/btcutil"
+)
+
+// MessageToStringClosure is used to produce less spammy log messages in trace
+// mode by setting the 'Curve" parameter to nil. Doing this avoids printing out
+// each of the field elements in the curve parameters for secp256k1.
+func MessageToStringClosure(msg lnwire.Message) logClosure {
```

Comment:
There is no original location for this function. The logic in `peer.go` remains the same and as far as `String()` function was removed from `lnwire` package now we need to recreate the logic of creation of description in every package.